Implement simple trips report

karls4 opened the pull request.

tananaev commented:
I feel like we need to remove code duplication between reports. Maybe split it into smaller components?

karls4 commented:
I will try remove the duplication.

tananaev commented:
Thanks. Happy to discuss the approach first if you have some ideas.

karls4 commented:
![image](https://user-images.githubusercontent.com/22108543/88718908-5d20b180-d11a-11ea-8d2d-4df84cd7103f.png)

What about having a report type picker?

tananaev commented:
You mean the last part? Yes, that makes sense. I also feel like there is an opportunity to generalize the table as well. Note that field for route report will be configurable in future, so maybe we can just have generic report table that takes a list of fields as an input?

karls4 commented:
I have added a url parameter to the report page to take the report type eg "route", "trips"
Added a report selector to the top of the report parameters sections
depending on what type of report is shown decides which table is shown

tananaev commented:
That's probably too much. I think we should still keep a page per report, but split sub-components into more generic ones.

tananaev commented:
Any updates?

karls4 commented:
Hi, no updates yet. I might get a chance this week to get a look at it.

karls4 commented:
![image](https://user-images.githubusercontent.com/22108543/89932202-2613cb00-dc05-11ea-935f-1f8d197bbc67.png)

What if I split this section into its own component and reuse it on the two report pages?

Do we need anything else added to it? 

Groups, show markers etc

tananaev commented:
That makes sense. We can add new stuff later in a separate PR. I would recommend keep PRs as small as possible and don't combine multiple changes.

tananaev commented:
We already have trips report.
